src/components/dialogs/ag-grid-table-rhf/custom-ag-grid-table.js

```diff
@@ -0,0 +1,308 @@
+        '& .ag-body-horizontal-scroll:not(.ag-scrollbar-invisible) .ag-horizontal-left-spacer:not(.ag-scroller-corner)':
+            {
+                visibility: 'hidden',
+            },
+
+        // hides right border for header of "Edit" column due to column being pinned
+
+        '& .ag-header-container': {
+            backgroundColor: theme.agGridBackground.color,
+        },
+        '& .ag-body': {
+            backgroundColor: theme.agGridBackground.color,
+        },
+        '& .ag-row': {
+            backgroundColor: theme.agGridBackground.color,
+        },
+        '& .ag-checkbox-input-wrapper': {
+            backgroundColor: theme.agGridBackground.color,
+        },
+        '& .ag-cell-edit-wrapper': {
+            height: 'inherit',
+            //border: 'none'
+        },
+    },
+    iconColor: {
+        color: theme.palette.primary.main,
+        justifyContent: 'flex-end',
+    },
+    selectedRow: {
+        backgroundColor: 'red',
+    },
+}));
+
+export const CustomAgGridTable = ({
+    name,
+    columnDefs,
+    defaultRowData,
+    fromCsvDataToFormValues, // this used to transform rows from csv file, to data can be displayed in the table
+    csvFileHeaders,
+    csvInitialData, // this is used if csv file generated has some row by default (comments for example)
+    ...props
+}) => {
+    const classes = useStyles();
+    const theme = useTheme();
+    const [gridApi, setGridApi] = useState(null);
+    const [selectedRows, setSelectedRows] = useState([]);
+    const [uploaderOpen, setUploaderOpen] = useState(false);
+    const { control } = useFormContext();
+    const [upDisabled, setUpDisabled] = useState(false);
+    const [downDisabled, setDownDisabled] = useState(false);
+    const intl = useIntl();
+    const { getValues } = useFormContext();
+
+    const { fields, append, remove, move, update } = useFieldArray({
+        control,
+        name: name,
+    });
+
+    // we add the unique generated id from react hook form to each row, so we can use it in getRowId
+    // the unique id used to identify the selected rows, so we can keep the selection when row index is changed.
+    const defaultRowsWithId = useMemo(() => {
+        const rows = getValues(name);
+        if (!rows) {
+            return [];
+        }
+        return getValues(name).map((v, i) => {
+            return {
+                id: fields[i].id,
+                ...v,
+            };
+        });
+    }, [fields, getValues, name]);
+
+    const [rowData, setRowData] = useState(defaultRowsWithId);
+
+    const handleAddRow = () => {
+        append(defaultRowData);
+        setRowData((oldValues) => {
+            const newVal = {
+                id: fields[fields.length - 1].id,
+                ...defaultRowData,
+            };
+            return [...oldValues, newVal];
+        });
+    };
+
+    const handleDeleteRows = () => {
+        selectedRows.forEach((val, index) => {
```

Comment:
I don't understand this method, why iterate on selectedRows and inside each iteration, use setSelectedRows ?